imageprocessing/src/main/java/com/example/imageprocessing/PreProcessing.java

```diff
@@ -0,0 +1,394 @@
+
+        double meanAngle = 0;
+        Log.d(TAG, "rows = " + lines.cols() + "\ncols = " + lines.cols());
+
+        //Analyzes the text line per line
+        for (int i = 0; i < lines.rows(); i++) {
+            //Get points from the beginning and the ending of the line of text
+            double[] vec = lines.get(i, 0);
+
+            //First point
+            double x1 = vec[0];
+            double y1 = vec[1];
+
+            //Second point
+            double x2 = vec[2];
+            double y2 = vec[3];
+
+            //Sum all the angle of the lines of text
+            meanAngle += Math.atan2(y2 - y1, x2 - x1);
+        }
+
+        //Calculate the meanAngle by dividing it with the number of rows
+        meanAngle /= lines.rows();
+
+        //Transform the angle in degrees
+        double degreesAngle = Math.toDegrees(meanAngle);
+        Log.i(TAG, "Mean angle=" + degreesAngle);
+        return degreesAngle;
+    }
+
+    /**
+     * Rotate the given bitmap with the given angle
+     * @param original The image that we want to corrected, must not be null
+     * @param degrees The degrees of the angle we want to rotate
+     * @return the rotatedImage
+     * @author Thomas Porro(g1), Giovanni Fasan (g1), Oscar Garrido (g1)
+     */
+    private static Bitmap rotateBitmap(@NonNull Bitmap original, double degrees) {
+
+        //Obtain the dimen of the image
+        int width = original.getWidth();
+        int height = original.getHeight();
+
+        //Prepare the rotation matrix. The minus before degrees allows us to rotate the
+        //image in the right way
+        Matrix matrix = new Matrix();
+        matrix.preRotate((float)-degrees);
+
+        //Rotate the Bitmap and returns it
+        return Bitmap.createBitmap(original, 0, 0, width, height, matrix, true);
+
+    }
+
+
+    /**
+     * Performs the skew correction
+     * @param image The image that I want to rotate, must not be null
+     * @return The image corrected
+     * @author Thomas Porro(g1), Giovanni Fasan (g1), Oscar Garrido (g1)
+     */
+    private Bitmap editSkew(@NonNull Bitmap image){
+        double angle = computeSkew(image);
+        return rotateBitmap(image, angle);
+    }
+
+
+    /**
+     * Detect if the image is bright
+     * @param imageMat the image we want to detect the brightness
+     * @return IMAGE_IS_OK if image is neither too bright nor too dark,
+     *         IMAGE_IS_BRIGHT if image is too bright,
+     *         IMAGE_IS_DARK if image is too dark.
+     * @author Thomas Porro(g1), Giovanni Fasan(g1), Leonardo Pratesi(g1)
+     */
+    private BrightnessValue isBright(Mat imageMat){
+        //Converts the image into a matrix
+        Mat brightnessMat = new Mat();
+
+        //Changes the format of the matrix into a RGB one
+        Imgproc.cvtColor(imageMat, brightnessMat, Imgproc.COLOR_RGBA2RGB);
+
+        //Obtain 3 different matrix with the 3 elemental colors
+        List<Mat> color = new ArrayList<>();
```

Comment:
I suggest to change this to imageColors. In this what I know that I'm working with the image that I want to work with and not with any other object containing colors